File: cc_Wallet/wallet_func.py
class WalletFunc:
    @staticmethod
    def add_pit_to_wallet(items, pit_type):
        messages = []
        if pit_type:
            items.append({'type': 'pit', 'pit_type': pit_type})
            messages.append(f'PIT ({pit_type}) added successfully to the wallet.')
        else:
            messages.append('Invalid PIT type. PIT not added to the wallet.')
        return {'messages': messages}

    @staticmethod
    def add_money_to_wallet(items, amount):
        messages = []
        if isinstance(amount, (int, float)) and amount > 0:
            items.append({'type': 'money', 'amount': amount})
            messages.append(f'Ar{amount:.2f} added successfully to the wallet.')
        else:
            messages.append('Invalid amount. Money not added to the wallet.')
        return {'messages': messages}

    @staticmethod
    def view_money_in_wallet(items):
        messages = []
        total_money = sum(item['amount'] for item in items if item['type'] == 'money')
        messages.append(f'Total Money in the wallet: Ar{total_money:.2f}')
        return {'messages': messages}

    @staticmethod
    def withdraw_money_from_wallet(items, amount):
        messages = []
        total_money = sum(item['amount'] for item in items if item['type'] == 'money')
        if isinstance(amount, (int, float)) and amount > 0 and amount <= total_money:
            items.append({'type': 'money', 'amount': -amount})
            messages.append(f'Ar{amount:.2f} withdrawn successfully from the wallet.')
        else:
            messages.append('Invalid amount. Money not withdrawn from the wallet.')
        return {'messages': messages}

File: cc_Wallet/WalletFunc.ts
import { WalletItem, WalletFuncResult } from './types';

export class WalletFunc {
    static addPITToWallet(items: WalletItem[], pitType: string): WalletFuncResult {
        const messages: string[] = [];

        if (pitType) {
            items.push({ type: 'pit', pit_type: pitType });
            messages.push(`PIT (${pitType}) added successfully to the wallet.`);
        } else {
            messages.push('Invalid PIT type. PIT not added to the wallet.');
        }

        return { messages };
    }

    static addMoneyToWallet(items: WalletItem[], amount: number): WalletFuncResult {
        const messages: string[] = [];

        if (typeof amount === 'number' && amount > 0) {
            items.push({ type: 'money', amount });
            messages.push(`Ar${amount.toFixed(2)} added successfully to the wallet.`);
        } else {
            messages.push('Invalid amount. Money not added to the wallet.');
        }

        return { messages };
    }

    static viewMoneyInWallet(items: WalletItem[]): WalletFuncResult {
        const messages: string[] = [];
        const totalMoney = items
            .filter(item => item.type === 'money')
            .reduce((sum, item) => sum + (item.amount || 0), 0);

        messages.push(`Total Money in the wallet: Ar${totalMoney.toFixed(2)}`);
        return { messages };
    }

    static withdrawMoneyFromWallet(items: WalletItem[], amount: number): WalletFuncResult {
        const messages: string[] = [];
        const totalMoney = items
            .filter(item => item.type === 'money')
            .reduce((sum, item) => sum + (item.amount || 0), 0);

        if (typeof amount === 'number' && amount > 0 && amount <= totalMoney) {
            items.push({ type: 'money', amount: -amount });
            messages.push(`Ar${amount.toFixed(2)} withdrawn successfully from the wallet.`);
        } else {
            messages.push('Invalid amount. Money not withdrawn from the wallet.');
        }

        return { messages };
    }
}
